src/main/groovy/io/openliberty/tools/gradle/tasks/DeployTask.groovy

```diff
@@ -682,8 +682,7 @@ class DeployTask extends AbstractServerTask {
                 File serverXML = new File(getServerDir(project).getCanonicalPath(), "server.xml")
 
                 try {
-                    scd = getServerConfigDocument(new CommonLogger(project), serverXML, server.configDirectory,
-                            server.bootstrapPropertiesFile, combinedBootstrapProperties, server.serverEnvFile, false, getLibertyDirectoryPropertyFiles(null))
+                    scd = getServerConfigDocument(new CommonLogger(project), getLibertyDirectoryPropertyFiles(null))
```

Comment:
You don't need to get serverXML anymore at line 682.

Reply:
Added back logic in getServerDocument() to compare server.xml